### PR TITLE
Fix entity spawning out of bounds

### PR DESCRIPTION
**Pull request by jorjic**
Fixes https://github.com/Facepunch/garrysmod-issues/issues/5028

### CODE REVIEW

File: garrysmod/gamemodes/sandbox/gamemode/commands.lua [hunk 1]
```diff
@@ -774,11 +774,31 @@ function Spawn_SENT( ply, EntityName, tr )
 		if ( !EntTable ) then return end
 
 		PrintName = EntTable.PrintName
+		
+		-- If the player is aimed at an entity, do not place below the entity
+		if ( !tr.HitWorld ) then
+			tr.HitNormal.z = 0
```
Comment (thegrb93) [inline garrysmod/gamemodes/sandbox/gamemode/commands.lua:780]: Modifying the trace data could screw up other things using the CanTool hook. Try this
```lua
local SpawnPos = tr.HitPos
if ( tr.HitWorld ) then
    SpawnPos  = SpawnPos + tr.HitNormal * 16
end
```

**Comment by jorjic** [comment]
This PR should mostly fix entities spawning out of bounds. With the changes, I can't easily make entities clip through the ground, and entities are now positioned better in tight spaces. It should be very close to the old behavior, otherwise.

I removed the part about moving the spawn position up by half of the entity's bounding box, because the return value for OBBCenter is 0 before spawning the entity. Maybe that can be re-added after Spawn(), but I didn't need it to fix the issue and I'm not confident that it won't cause problems for some other entities.

File: garrysmod/gamemodes/sandbox/gamemode/commands.lua (hunk 1 above)
Comment (jorjic) [reply garrysmod/gamemodes/sandbox/gamemode/commands.lua]: Thank you!

**Comment by jorjic** [comment]
I found a simpler way to fix this that is also closer to the old behavior. Unless something else comes up, this should be the final commit. Here is a demonstration

novyy

**Pull request merged**